### PR TITLE
Add new DonorsChoose donation flow

### LINKED ISSUE #481: add more robust donation information to donation flow
https://docs.google.com/document/d/1CNsINZ-aj6zSf3EmJqxOCsRizxEXrGDHm1JykD8yCa0/edit#

### PR DESCRIPTION
**Pull request by tongxiang**
#### What's this PR do?
It changes the DonorsChoose donation flow to match [this new flow](https://docs.google.com/document/d/1CglUNrQ-T93ubuAZ0kt7Sm3otwTPnZ3TLGZF_PoEu70/edit#). 

A few important changes have been made, code-wise:
1. The `findProject` method is now called by the `start` method. 
2. The `donorschooseConfigModel.js` has been changed to reflect new params. 
3. General refactoring to support the new order of messages and new param names. 

#### Where should the reviewer start?
`DonorsChooseDonationaController.js`. 

#### How should this be manually tested?
[This Postman collection](https://www.getpostman.com/collections/b6a81618402acc8c598c) should help us test it. 
For some reason, however, the test API doesn't provide results which can be successfully donated to. Once we get the updated credentials from DonorsChoose, we'll test this flow again to confirm. 

#### What are the relevant tickets?
Closes #481.

### CODE REVIEW

File: app/lib/donations/models/donorschooseConfigModel.js [hunk 1]
```diff
@@ -12,33 +12,21 @@ var donorschooseConfigSchema = new mongoose.Schema({
...
+  max_donations_reached_oip: Number,
 
-  // Error notification OIP. 
-  error_direct_user_to_restart: Number,
-
-  // Maximum number of donations a user can make for this campaign.
-  max_donations_allowed: Number,
-
-  // OIP sent when the maximum number of donations has already been sent.
-  max_donations_reached_oip: Number
```
Comment (jonuy) [inline app/lib/donations/models/donorschooseConfigModel.js]: Any particular reason we're removing all the comments here?

File: app/lib/donations/controllers/DonorsChooseDonationController.js [hunk 2]
```diff
@@ -126,26 +127,12 @@ DonorsChooseDonationController.prototype.start = function(request, response) {
...
-  if (typeof request.query.id === 'undefined'
-      || typeof request.body.mobile === 'undefined'
-      || typeof request.body.location === 'undefined') {
-    response.status(406).send('Missing required params.');
-    return;
-  }
+DonorsChooseDonationController.prototype.findProject = function(mobileNumber, configId) {
 
-  var config = app.getConfig(app.ConfigName.DONORSCHOOSE, request.query.id);
+  var config = app.getConfig(app.ConfigName.DONORSCHOOSE, 201);
```
Comment (jonuy) [inline app/lib/donations/controllers/DonorsChooseDonationController.js]: Hardcoding the `201` here? This seemed ok coming from the `request.query.id`.

File: app/lib/donations/controllers/DonorsChooseDonationController.js [hunk 3]
```diff
@@ -126,26 +127,12 @@ DonorsChooseDonationController.prototype.start = function(request, response) {
...
-  // and assigning query params accordingly. 
-  if (parseInt(request.body.location)) {
-    var locationFilter = 'keywords=' + request.body.location; // If zip. 
-  }
-  else {
-    var locationFilter =  'state=' + request.body.location; // If state. 
-  }
-
-  // Subject code for all 'Math & Science' subjects.
+    // Subject code for all 'Math & Science' subjects.
```
Comment (jonuy) [inline app/lib/donations/controllers/DonorsChooseDonationController.js]: Spacing/tabbing is off

**Comment by jonuy** [comment]
`npm test` passed on my machine. Testing through Postman also looked good. I did get the error you were mentioning though with the project returned from the DonorsChoose API. Have you already sent a note to Michaela or DonorsChoose about this? Otherwise, I can help with that.

Just a few notes above on the code. Otherwise, looks good to merge when you're ready.

File: app/lib/donations/models/donorschooseConfigModel.js (hunk 1 above)
Comment (tongxiang) [inline app/lib/donations/models/donorschooseConfigModel.js]: This was me abiding by the Objective-C principle that variable names should be self-documenting. 

I realized that the schema property names were descriptive enough that I felt that we could take them out. 

Happy to put them back in, though, if you disagree!

**Comment by tongxiang** [comment]
Thanks, @jonuy! I'm going to hold off on merging this in until DonorsChoose refills the credits in their QA transactions API and we can confirm again that the transaction works.

File: app/lib/donations/models/donorschooseConfigModel.js (hunk 1 above)
Comment (jonuy) [inline app/lib/donations/models/donorschooseConfigModel.js]: Gotcha, makes sense

**Pull request merged**